docs/src/wallet_to_wallet_with_tariscript.md

```diff
@@ -0,0 +1,205 @@
+| Rangeproof    | \\( RP_c \\) for \\( \hat{C}_c \\)            |
+| Script Hash   | \\( \sigma_c = H( \alpha_c) \\)               |
+| Script offset public key | \\( K_{Oc} \\)                     |
+
+| Transaction Kernel |       |
+|--------------------|-------|
+| Public Excess      | \\( X_s + K_b \\)    |
+| Signature          | \\( s, R \\)         |
+| Fee                | f                    |
+| Metadata           | m                    |
+
+Alice will now send this finalized transaction back to Bob so he has a copy of it and the script and both parties can now submit this transaction to the blockchain.
+
+### Bob after transaction broadcast
+
+After the transaction has been broadcast to the mempool Bob will have to monitor the blockchain to detect what height the transaction is mined at. Bob will then
+record this Height in his database for use when spending the UTXO.
+
+## Changes to Transaction messages and store data in Rust
+In [RFC](https://rfc.tari.com/RFC-0201_TariScript.html) the changes to the TransactionInput, TransactionOutput and Transaction structs are detailed but there are
+two more structs that will need to be updated to support TariScript.
+
+This is the struct that is sent by the Sender to the Recipient and needs to contain 2 new fields
+```rust,ignore
+pub struct SingleRoundSenderData {
+    /// The transaction id for the recipient
+    pub tx_id: u64,
+    /// The amount, in µT, being sent to the recipient
+    pub amount: MicroTari,
+    /// The offset public excess for this transaction
+    pub public_excess: PublicKey,
+    /// The sender's public nonce
+    pub public_nonce: PublicKey,
+    /// The transaction metadata
+    pub metadata: TransactionMetadata,
+    /// Plain text message to receiver
+    pub message: String,
+    
+    // NEW FIELDS
+    
+    /// Hash of the receivers UTXO script, \sigma
+    pub script_hash: Vec<u8,
+    /// Public script offset key chosen for Recipient, K_o
+    pub public_script_offset_key: PublicKey,
+```
+
+And the receiver will need to keep track of their spendable UTXO's using the following updated struct
+
+```rust,ignore
+pub struct UnblindedOutput {
+    pub value: MicroTari,
+    pub spending_key: BlindingFactor,
+    pub features: OutputFeatures,
+    
+    // NEW FIELDS
+    
+    /// The serialised script
+    script: Vec<u8>,
+    /// The script input data, if any
+    input_data: Vec<u8>,
+    /// The block height that the UTXO was mined
+    height : u64,
+    /// Script private key, k_s
+    script_private_key: PrivateKey,
+}
+```
+**NOTE:** The height in the previous struct will need to be updated AFTER the transaction negotiation is complete once the Receiver has detected the UTXO has been mined.
+
+
+## Questions
+1. Does the receiver need to get the script itself earlier than finding it in the Finalized transaction?
+2. Do parties agree of the Output Features of the received output or does Bob just choose it? Currently it's assumed default
```

Comment:
I think we can leave this as assumed default as in most cases this does not matter.  But the mere fact that it's in the tx and signed by the script and in the total script offset at least the person who signs the script_offset needs to agree on it. If both parties sign it, they both need to agree to it. In most cases, this will only be the sender.

Reply:
Ja, I think in the implementation we can assume default for now because only Coinbases currently use a non-default value but will leave it in this Doc description for a time when that is not the case.